### PR TITLE
build: use the correct base container for armhf/aarch64

### LINKED ISSUE #613: The container build script should choose the right base image for the right arch
Currently `alpine` (amd64) is hard-coded in the build script, which makes the arm64 image not work.

@bassam could just quickly upstream a part of his WIP here to fix this: https://github.com/bassam/rook/commit/b214910bc55aeb1fda1ae76dc648fe5308ba92ac

### PR DESCRIPTION
**Pull request by bassam**
Fixes #613

### CODE REVIEW

File: build/release/containers.sh [hunk 1]
```diff
@@ -34,7 +34,7 @@ build_artifact() {
     local name=$(get_image_name $os $arch $repo ${RELEASE_VERSION})
 
     echo building docker container ${name}
-    docker build -t ${registry}${name} $tmpdir
+    docker build --no-cache -t ${registry}${name} $tmpdir
```
Comment (luxas) [inline build/release/containers.sh]: nit: Common practice in k8s is to pass `--pull` to every build

Comment (bassam) [inline build/release/containers.sh]: ok will fix.

**Comment by bassam** [comment]
jenkins failing with:
```
 ---> Running in 231b407d91e7
fetch http://dl-cdn.alpinelinux.org/alpine/v3.5/main/x86_64/APKINDEX.tar.gz
[91mpanic: standard_init_linux.go:178: exec user process caused "exec format error" [recovered]
	panic: standard_init_linux.go:178: exec user process caused "exec format error"
```

We might need to upgrade the jenkins AMI to support a 4.8 kernel which will use qemu to build the image correctly. This works on my machine.

**Comment by luxas** [comment]
@bassam or we could simply do QEMU emulation inside of the container, works on any 4.x kernel (at least): https://github.com/kubernetes/community/blob/master/contributors/design-proposals/multi-platform.md

**Comment by bassam** [comment]
@luxas what I don't like about emulation inside the container is that qemu ends being shipped inside the container. I modified the Jenkins slave to use binfmt. All good now

File: build/README.md [hunk 2]
```diff
@@ -15,6 +15,52 @@ A capable machine (2+ cores, 8+ GB of memory) with Docker installed locally:
...
+RemainAfterExit=yes
+
+[Install]
+WantedBy=multi-user.target
+EOF
+
+systemctl enable update-binfmt.service
+```
+
+To test run the following:
```
Comment (jbw976) [inline build/README.md]: what's expected to be shown in the output from these tests commands?

Comment (bassam) [inline build/README.md]: That the complete correctly, no errors

**Pull request merged**